src/hayhooks/server/pipelines/utils.py

```diff
@@ -613,6 +642,15 @@ async def _stream_chunks_from_queue(
             if exception is not None:
                 raise exception
 
+        # Process items from external queue first (non-blocking)
+        if external_event_queue is not None:
+            while not external_event_queue.empty():
+                try:
+                    external_item = external_event_queue.get_nowait()
+                    yield external_item
```

Comment:
should we check this event is not something unexpected, more for surfacing/debugging than anything else? i.e add
```
def _validate_external_event(event: Any) -> bool:
    return isinstance(event, (StreamingChunk, OpenWebUIEvent, str, dict))
```
and invoke here. If fails log warn, something like that